appinventor/components-ios/src/ProximitySensor.swift

```diff
@@ -0,0 +1,117 @@
+// -*- mode: swift; swift-mode:basic-offset: 2; -*-
+// Copyright 2023 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+import Foundation
+import UIKit
+import CoreMotion
+
+class ProximitySensor: NonvisibleComponent {
+  private var motionManager: CMMotionManager?
+  private var enabled: Bool = true
+  private var distance: Float = 0.0
+  private var keepRunningWhenOnPause: Bool = false
+  override init(_ container: ComponentContainer) {
+    super.init(container)
+    // Check if the proximity sensor is available
+    let device = UIDevice.current
+    device.isProximityMonitoringEnabled = true
+    
+    // Register for proximity state changes
+    NotificationCenter.default.addObserver(self, selector: #selector(proximityStateChanged(_:)), name: UIDevice.proximityStateDidChangeNotification, object: device)
+    
+    // Configure motion manager for maximum range
+    motionManager = CMMotionManager()
+    motionManager?.startDeviceMotionUpdates()
+    
+    startListening()
```

Comment:
See my comments on the other files about the interaction between the init function, Initialize, and the Enabled property.